Submitted By:     Adam Zlehovszky <zleho@index.hu>
Date:             2004.02.09.
Initial Packages: 2.9.5
Description:      This patch fixes an annoying bug of pacman.
                  Always extract config files as .pacnew, instead of overwriting
                  the users's configs.

--- pacman-2.9.5.orig/src/pacman.c	2005-01-12 00:14:16.000000000 +0100
+++ pacman-2.9.5/src/pacman.c	2005-02-19 13:30:43.989002328 +0100
@@ -1885,13 +1885,13 @@
 						installnew = 1;
 					} else {
 						char newpath[PATH_MAX];
-						vprint("  action: saving current file and installing new one\n");
-						installnew = 1;
-						snprintf(newpath, PATH_MAX, "%s.pacsave", expath);
-						if(rename(expath, newpath)) {
-							logaction(stderr, "error: could not rename %s: %s", expath, strerror(errno));
+						vprint("  action: keeping current file and installing new one with .pacnew ending\n");
+						installnew = 0;
+						snprintf(newpath, PATH_MAX, "%s.pacnew", expath);
+						if(copyfile(temp, newpath)) {
+							logaction(stderr, "error: could not install %s as %s: %s", expath, newpath, strerror(errno));
 						} else {
-							logaction(stderr, "warning: %s saved as %s", expath, newpath);
+							logaction(stderr, "warning: %s installed as %s", expath, newpath);
 						}
 					}
 
